### PR TITLE
Do NOT allow termvectors on nested fields

### LINKED ISSUE #21625: Term vectors query returns nothing in nested inner objects if term vectors are stored in the index
**Elasticsearch version**: 5.0.0, 2.4.1

**Plugins installed**: []

**JVM version**: 1.8.0_112-b15 64-bit

**OS version**: Windows 10 v. 1607 (build 14393.447)

**Description of the problem including expected versus actual behavior**:

When making a `_termvectors` query to an object in a `"type": "nested"` field, response is returned with `"term_vectors" : { }` if vectors are stored (`"store": true` in mapping). If the object isn't nested or term vectors aren't stored, the object returned is filled properly (examples below). When TV storing is enabled then despite the fact that I'm unable to retrieve them, highlighting seems to be significantly faster, as if term vectors were present.

**Steps to reproduce**:
1. `PUT /mcve`

```
2.  PUT /mcve/ex/_mapping
    {
      "properties": {
        "parent": {
          "type": "nested",
          "properties": {
            "value": {"type": "text", "term_vector": "with_positions_offsets", "store": true}
          }
        }
      }
    }
```

```
3.  PUT /mcve/ex/ample
    {
      "parent": {
        "value": "hello world"
      }
    }
```

```
4.  PUT /mcve/ex/ample/_termvectors
    {
      "fields": ["parent.value"]
    }
```

Response:

```
{
  "_index" : "mcve",
  "_type" : "ex",
  "_id" : "ample",
  "_version" : 1,
  "found" : true,
  "took" : 1,
  "term_vectors" : { }
}
```

However, if I disable storing term vectors by changing this line in mapping:

    "value": {"type": "text", "term_vector": "with_positions_offsets", "store": true}

to this:

    "value": {"type": "text"}

the result will be:

```
{
  "_index" : "mcve",
  "_type" : "ex",
  "_id" : "ample",
  "_version" : 1,
  "found" : true,
  "took" : 60,
  "term_vectors" : {
    "parent.value" : {
      "field_statistics" : {
        "sum_doc_freq" : 2,
        "doc_count" : 1,
        "sum_ttf" : 2
      },
      "terms" : {
        /* ...snip... */
      }
    }
  }
}
```

But highlighting queries will be slower (tested on a ~80 MB index with ~600 documents), as if term vectors were in fact built previously.

**Provide logs (if relevant)**:

--

[Related SO question](http://stackoverflow.com/q/40527963/1937994)

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Requesting _termvectors on a nested field or any sub-fields of a nested field
returns empty results.

Closes #21625, #32652

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java [hunk 1]
```diff
@@ -169,6 +170,15 @@ private static boolean isValidField(MappedFieldType fieldType) {
         if (fieldType.indexOptions() == IndexOptions.NONE) {
             return false;
         }
+        // and must not be under nested field
```
Comment (jimczi) [inline server/src/main/java/org/elasticsearch/index/termvectors/TermVectorsService.java]: We should check all levels in the path, for instance in `foo.bar.baz`, `foo` can be a simple object field and `foo.bar` a nested field.

**Review by jimczi — CHANGES_REQUESTED** [review]
Thanks @mayya-sharipova , I left one comment

**Review by jimczi — APPROVED** [review]
LGTM, thanks @mayya-sharipova

**Comment by BurntSushi** [comment]
@mayya-sharipova Do you happen to know why this change was reverted? As far as I can tell, termvectors still doesn't work with nested documents.

**Comment by mayya-sharipova** [comment]
@BurntSushi  We were a little hasty in disabling `_termvectors` request on nested docs, because this request works when term vectors are not explicitly stored, and disabling this request all together is a breaking change. That's why we had to revert the PR. 
You are right, the request `_termvectors` still doesn't work if term vectors were explicitly stored and pre-calculated during indexing.

**Comment by BurntSushi** [comment]
@mayya-sharipova Great, thanks for the clarification!

**Pull request merged**